packages/compile/src/generate/code-gen.js

```diff
@@ -17,14 +17,7 @@ let codeGenerator = (parsedModel, opts) => {
   // Set to 'decl', 'init-lookups', 'eval', etc depending on the section being generated.
   let mode = ''
   // Set to true to output all variables when there is no model run spec.
-  let outputAllVars
-  if (spec.outputVars && spec.outputVars.length > 0) {
-    outputAllVars = false
-  } else if (spec.outputVarNames && spec.outputVarNames.length > 0) {
-    outputAllVars = false
-  } else {
-    outputAllVars = true
-  }
+  let outputAllVars = spec.outputVarNames === undefined || spec.outputVarNames.length === 0
```

Comment:
I simplified this logic now that the user can only provide `outputVarNames` and `outputVars` is only used internally.